File: ScorpioProto/CSharp/Example/src/TableTest.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Scorpio.Conversion;

namespace Datas {
    public partial class TableTest : ITable {
        const string FILE_MD5_CODE = "f07d3fff17de6b37025a951b272f6c4c";
        private int m_count = 0;
        private Dictionary<int, DataTest> m_dataArray = new Dictionary<int, DataTest>();
        public TableTest Initialize(string fileName, IReader reader) {
            var row = reader.ReadInt32();
            var layoutMD5 = reader.ReadString();
            if (layoutMD5 != FILE_MD5_CODE) {
                throw new Exception("File schemas do not match [TableTest] : " + fileName);
            }
            ConversionUtil.ReadHead(reader);
            for (var i = 0; i < row; ++i) {
                var pData = DataTest.Read(fileName, reader);
                if (m_dataArray.TryGetValue(pData.ID(), out var value))
                    value.Set(pData);
                else
                    m_dataArray[pData.ID()] = pData;
            }
            m_count = m_dataArray.Count;
            return this;
        }
        public DataTest GetValue(int ID) {
            if (m_dataArray.TryGetValue(ID, out var value))
                return value;
            throw new Exception($"TableTest not found data : {ID}");
        }
        public bool Contains(int ID) {
            return m_dataArray.ContainsKey(ID);
        }
        public Dictionary<int, DataTest> Datas() {
            return m_dataArray;
        }
        public IData GetValueObject(object ID) {
            return GetValue((int)ID);
        }
        public bool ContainsObject(object ID) {
            return Contains((int)ID);
        }
        public IDictionary GetDatas() {
            return Datas();
        }
        public int Count() {
            return m_count;
        }
    }
}

File: ScorpioProto/Nodejs/src/TableTest.js
//本文件为自动生成，请不要手动修改

const DataTest = require("./DataTest")
class TableTest {
    constructor() {
        this.m_count = 0;
        this.m_dataArray = new Map();
    }
    Initialize(fileName, reader) {
        let row = reader.ReadInt32();
        if ("f07d3fff17de6b37025a951b272f6c4c" != reader.ReadString()) {
            throw new Error("File schemas do not match [TableTest] : ${fileName}");
        }
        reader.ReadHead();
        for (let i = 0; i < row; i ++) {
            let pData = new DataTest(fileName, reader);
            if (this.m_dataArray.has(pData.ID)) {
                this.m_dataArray.get(pData.ID).Set(pData);
            } else {
                this.m_dataArray.set(pData.ID, pData);
            }
        }
        this.m_count = this.m_dataArray.size;
        return this;
    }
    GetValue(ID) {
        return this.m_dataArray.get(ID)
    }
    Contains(ID) {
        return this.m_dataArray.has(ID)
    }
    Datas() {
        return this.m_dataArray
    }
    GetValueObject(ID) {
        return this.GetValue(ID)
    }
    ContainsObject(ID) {
        return this.Contains(ID)
    }
    Count() {
        return this.m_count;
    }
}
module.exports = TableTest;